tests/unittest_observer/FaceliftObserverTest.cpp

```diff
@@ -0,0 +1,76 @@
+#include <gtest/gtest.h>
+#include "IPCProxyBase.h"
+#include "InterfaceBase.h"
+#include <QSignalSpy>
+
+namespace {
+
+using namespace facelift;
+
+class Counter
+{
+public:
+    Counter() { m_value = 0; }
+
+    void incValue() {
+        m_value++;
+    }
+    int getValue() const {
+       return m_value;
+    }
+private:
+    int m_value;
+};
+
+class IPCProxyBaseTest : public ::testing::Test
+{
+public:
+    Counter c1;
+    Counter c2;
+    StandartObserver  < std::function<void()> > * obs1 = new StandartObserver  < std::function<void()> > (std::bind(&Counter::incValue, &c1) );
```

Comment:
use auto

Reply:
Unfortunately auto is not supported.
In this case, we have some limitation knowing as AAA Style (Almost Always Auto):
https://herbsutter.com/2013/08/12/gotw-94-solution-aaa-style-almost-always-auto/

Reply:
agree
**ISO C++ forbids in-class initialization of non-const static member** 
it was misleading, thought i in some function.

Reply:
please make this line less long, 
e.g. this thing "> >" is not needed anymore

StandartObserver  < 
one space is spare

delete extra spaces in all this line like this:
`>>(std::bind(&Counter::incValue, &c1)); `

you could use using keyword as well
and as mentioned earlier use unique_ptr or class member.